src/services/chatgpt.ts

```diff
@@ -0,0 +1,88 @@
+const BASE_URL = 'https://api.openai.com/v1'
+const apiKey = '11111'
+
+interface EventDetails {
+  summary: string
+  location: string
+  start: {
+    dateTime: Date
+    timeZone: string
+  }
+  end: {
+    dateTime: Date
+    timeZone: string
+  }
+  description: string
+}
+
+type GetEventDetailsSuccessResponse = {
+  id: string
+  object: string
+  created: number
+  model: string
+  choices: {
+    message: {
+      role: string
+      content: string
+    }
+    finish_reason: string
+    index: number
+  }[]
+}
+
+type GetEventDetailsErrorResponse = {
+  isOk: false
+  data: null
+  error: {
+    message: string
+    type: string
+    param: null
+    code: string
+  }
+}
+
+type GetEventDetailsResponse = GetEventDetailsSuccessResponse | GetEventDetailsErrorResponse
+
+export async function getEventDetails(eventData: string): Promise<any> {
+  try {
+    const response = await fetch(`${BASE_URL}/completions`, {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${apiKey}`,
+      },
+      body: JSON.stringify({
+        model: 'text-davinci-003',
+        prompt: `extract event details in google calendar api json format from this text: ${eventData}`,
+        temperature: 0,
+        max_tokens: 100,
+        top_p: 1,
+        frequency_penalty: 0.0,
+        presence_penalty: 0.0,
+        stop: ['\n'],
+      }),
+    })
```

Comment:
Can refactor using axios using
```
    const response = await axios.post<GetEventDetailsResponse>(
      `${BASE_URL}/completions`,
      {
        model,
        prompt: `extract event details in google calendar api json format from this text: ${eventData}`,
        temperature: 0,
        max_tokens: 100,
        top_p: 1,
        frequency_penalty: 0.0,
        presence_penalty: 0.0,
        stop: ['\n'],
      },
      {
        headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          Authorization: `Bearer ${apiKey}`,
        },
      }
    );
```

Reply:
what is the advantage of using axios in this case?